src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -1104,16 +1104,16 @@ public static bool IsDedicatedAdminConnection(ConnectionDetails connectionDetail
         /// Build a connection string from a connection details instance
         /// </summary>
         /// <param name="connectionDetails"></param>
-        public static string BuildConnectionString(ConnectionDetails connectionDetails)
+        public static string BuildConnectionString(ConnectionDetails connectionDetails, Boolean isXELite = false)

```

Comment:
we should use C# primitives rather than .Net BCL classes when we have a choice.  i.e. please use `bool` here per .Net Framework coding guidelines.

Also, not sure the parameter name is ideal.  We're introducing a coupling between profiler and connectionservice that's inverted.  i.e., profilerservice can know about connectionservice but connectionservice shouldn't know about profilerservice.  Is there a different parameter name that more generically describes the customizations you're trying to make to the connection string for use is XELite?  And could you explain a bit why these customizations are needed?

Reply:
XELite does not accept certain options in the connection string and won't function (IE, no data will be streamed) when a string containing them is passed as a parameter to the XELiveEventStreamer. This boolean flag is to used to omit those problematic options from the connection string when it is being generated so that it works properly. I'll name it to something else and change the Boolean to bool.